libsolidity/AST.cpp

```diff
@@ -152,6 +153,50 @@ void ContractDefinition::checkAbstractFunctions()
 		}
 }
 
+void ContractDefinition::checkAbstractConstructors()
+{
+	set<FunctionDefinition const*> argumentsNeeded;
+	set<FunctionDefinition const*> argumentsProvided;
+	// check that we get arguments for all base constructors that need it.
+	// If not mark the contract as abstract (not fully implemented)
+	vector<ContractDefinition const*> const& bases = getLinearizedBaseContracts();
+	for (ContractDefinition const* contract: bases)
+	{
+		FunctionDefinition const* constructor = contract->getConstructor();
+		if (constructor)
+		{
+			if (!constructor->getParameters().empty())
+				argumentsNeeded.insert(constructor);
+			for (auto const& modifier: constructor->getModifiers())
+			{
+				auto baseContract = dynamic_cast<ContractDefinition const*>(
+					modifier->getName()->getReferencedDeclaration());
+				if (baseContract)
+				{
+					FunctionDefinition const* baseConstructor = baseContract->getConstructor();
+					if (argumentsNeeded.count(baseConstructor) == 1)
```

Comment:
baseConstructor might be nullptr - we should not insert that into argumentsProvided. Please add a test for that case.